httpapi/coreapi/documents_api.go

```diff
@@ -0,0 +1,89 @@
+package coreapi
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/centrifuge/go-centrifuge/documents"
+	"github.com/centrifuge/go-centrifuge/utils/httputils"
+	"github.com/go-chi/chi"
+	"github.com/go-chi/render"
+	logging "github.com/ipfs/go-log"
+)
+
+var log = logging.Logger("core_api")
+
+type handler struct {
+	srv           Service
+	tokenRegistry documents.TokenRegistry
+}
+
+// CreateDocument creates a document.
+// @summary Creates a new document and anchors it.
+// @description Creates a new document and anchors it.
+// @id create_document
+// @tags Documents
+// @accept json
+// @param authorization header string true "centrifuge identity"
+// @param body body coreapi.CreateDocumentRequest true "Document request"
+// @produce json
+// @Failure 500 {object} httputils.HTTPError
+// @Failure 400 {object} httputils.HTTPError
+// @Failure 403 {object} httputils.HTTPError
+// @success 200 {object} coreapi.DocumentResponse
+// @router /documents [post]
+func (h handler) CreateDocument(w http.ResponseWriter, r *http.Request) {
+	var err error
+	var code int
+	defer func() {
```

Comment:
This looks like a common block for all handlers, do you want to extract it so it can be reused?

Reply:
not unless we want to deal with pointers. ATM, i'll keep it as is.